internal/locate/region_request.go

```diff
@@ -1776,7 +1776,7 @@ func (s *RegionRequestSender) onSendFail(bo *retry.Backoffer, ctx *RPCContext, r
 		return errors.WithStack(err)
 	} else if LoadShuttingDown() > 0 {
 		return errors.WithStack(tikverr.ErrTiDBShuttingDown)
-	} else if errors.Cause(err) == context.DeadlineExceeded && req.MaxExecutionDurationMs < uint64(client.ReadTimeoutShort.Milliseconds()) {
+	} else if (errors.Cause(err) == context.DeadlineExceeded || strings.Contains(err.Error(), "context deadline exceeded")) && req.MaxExecutionDurationMs < uint64(client.ReadTimeoutShort.Milliseconds()) {
```

Comment:
Need to add explanations here as the conditions are complex, it's better to abstract a function for kv timeout related check.

Reply:
done.